torchx/schedulers/ray/ray_driver.py

```diff
@@ -130,24 +139,43 @@ def main() -> None:  # pragma: no cover
     actors: List[RayActor] = load_actor_json("actors.json")
     # pyre-fixme[16]: Module `worker` has no attribute `init`.
     ray.init(address="auto", namespace="torchx-ray")
-    pg: PlacementGroup = create_placement_group(actors)
-    command_actors: List[CommandActor] = create_command_actors(actors, pg)
 
-    active_workers = [
-        command_actor.exec_module.remote()  # pyre-ignore
-        for command_actor in command_actors
-    ]
+    pending_placement_groups = [
+        create_placement_group(actors[i : i + 1]) for i in range(len(actors))
+    ]  # trace all the placement groups
+    active_tasks = [
+        pg.ready() for pg in pending_placement_groups
+    ]  # tasks of creating all required placement groups
 
+    result: Optional[
+        PlacementGroup
+    ]  # result from a completed task, either a command execution result None or a created placement group
     # Await return result of remote ray function
-    while len(active_workers) > 0:
-        _logger.info(f"running ray.wait on {active_workers}")
+    while len(active_tasks) > 0:
+
+        _logger.info(f"running ray.wait on {active_tasks}")
 
         # pyre-fixme[16]: Module `worker` has no attribute `wait`.
-        completed_workers, active_workers = ray.wait(active_workers)
+        completed_tasks, active_tasks = ray.wait(active_tasks)
         # If a failure occurs the ObjectRef will be marked as completed.
         # Calling ray.get will expose the failure as a RayActorError.
-        for object_ref in completed_workers:
-            ray.get(object_ref)
+        for object_ref in completed_tasks:
+            # completed tasks contains two kinds of tasks:
+            # 1) placement group creation; 2) command actor execution
+            result = ray.get(object_ref)  # pyre-ignore
+            if isinstance(result, PlacementGroup):
+                new_actor: CommandActor = create_command_actors(
+                    actors[
+                        0:1
+                    ],  # use actors[0:1] since every actor in the list is the same
```

Comment:
this isn't guaranteed for heterogeneous jobs with multiple roles

Reply:
done, now it's using a dictionary which uses PlacementGroup Id as the key and actor index as the value to match the placement groups and the actors.